### PR TITLE
CustomRoles SpawnChance Feature

### PR DESCRIPTION
**Pull request by xNexus-ACS**
Suggested in Discord

### CODE REVIEW

**Review by louis1706 — APPROVED** [review]
Tested by xNexus-ACS and don't see any problem

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 1]
```diff
@@ -748,6 +755,16 @@ protected virtual void RoleRemoved(Player player)
         {
         }
 
+        private void OnInternalSpawning(SpawningEventArgs ev)
+        {
+            if (SpawnChance > 0 && !Check(ev.Player) && ev.Player.Role.Type == Role)
+            {
+                int r = Loader.Random.Next(100);
+                if (r <= SpawnChance)
+                    AddRole(ev.Player);
```
Comment (iopietro) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:764]:
```suggestion
            if (SpawnChance > 0 && !Check(ev.Player) && ev.Player.Role.Type == Role)
            {
                if (Loader.Random.Next(100) <= SpawnChance)
                    AddRole(ev.Player);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.CustomRoles/API/Features/CustomRole.cs (hunk 1 above)
Comment (xNexus-ACS) [reply Exiled.CustomRoles/API/Features/CustomRole.cs]: Done

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 2]
```diff
@@ -748,6 +755,15 @@ protected virtual void RoleRemoved(Player player)
         {
         }
 
+        private void OnInternalSpawning(SpawningEventArgs ev)
+        {
+            if (SpawnChance > 0 && !Check(ev.Player) && ev.Player.Role.Type == Role)
+            {
+                if (Loader.Random.Next(100) <= SpawnChance)
+                    AddRole(ev.Player);
+            }
```
Comment (iopietro) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:764]:
```suggestion
            if (SpawnChance > 0 && !Check(ev.Player) && ev.Player.Role.Type == Role && Loader.Random.Next(100) <= SpawnChance)
                AddRole(ev.Player);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.CustomRoles/API/Features/CustomRole.cs (hunk 2 above)
Comment (xNexus-ACS) [reply Exiled.CustomRoles/API/Features/CustomRole.cs]: Done XD

**Review by iopietro — APPROVED** [review]

**Comment by louis1706** [comment]
Update branch

**Comment by xNexus-ACS** [comment]
> Update branch

Done

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**